app/models/recipe.rb

```diff
@@ -1,3 +1,13 @@
 class Recipe < ApplicationRecord
    validates_presence_of :name, :complexity, :genre
+   has_many :recipe_ingredients
+   has_many :ingredients, through: :recipe_ingredients
+
+   def total_cost
+      total = 0
+      ingredients.each do |ingredient|
```

Comment:
You can do this with AR instead! When you find yourself doing `.each` in a model method, there's generally an ActiveRecord method that could do it instead. 

Consider: `ingredients.sum(:cost)`